Source/Core/VideoCommon/UberShaderCommon.cpp

```diff
@@ -0,0 +1,26 @@
+// Copyright 2017 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include "VideoCommon/UberShaderCommon.h"
+#include "VideoCommon/VideoConfig.h"
+
+namespace UberShader
+{
+void WriteUberShaderCommonHeader(ShaderCode& out, APIType api_type)
+{
+  // ==============================================
+  //  BitfieldExtract for APIs which don't have it
+  // ==============================================
+  if (!g_ActiveConfig.backend_info.bSupportsBitfield)
+  {
+    out.Write("uint bitfieldExtract(uint val, int off, int size) {\n"
```

Comment:
If it's just a mesa issue, and all the drivers that support es3 support GL_MESA_shader_integer_functions, we could use that extension version of bitfieldExtract instead